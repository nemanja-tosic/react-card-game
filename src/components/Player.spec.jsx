// @flow
import React from 'react';
import { shallow } from 'enzyme';

import PlayerComponent from './Player';
import { TestPlayer } from '../test/support/test-player';
import { TestCards } from '../test/support/test-cards';

describe('PlayerComponent', () => {
  let component;
  let cardClickedSpy;
  let cardRefSpy;
  let player;


  beforeEach(() => {
    player = TestPlayer.create({
      id: 1,
      name: 'John',
      cards: TestCards.createHand(),
      takenCards: [],
      score: 0,
      isActive: true,
      isUserControlled: true,
    });
    cardClickedSpy = jest.fn();
    cardRefSpy = jest.fn();

    component = shallow(
      <PlayerComponent
        player={player}
        slot={0}
        cardRef={cardRefSpy}
        onCardClicked={cardClickedSpy}
      />
    );
  });

  it('should be marked as active when active', () => {
    player.isActive = true;

    component.setProps({player})

    expect(component.getNode()).toMatchSnapshot();
  });

  it('should be marked as inactive when inactive', () => {
    player.isActive = false;

    component.setProps({player})

    expect(component.getNode()).toMatchSnapshot();
  });

  it('should have the cards in the portrait orientation when slot is 0', () => {
    component.setProps({slot: 0});

    expect(component.getNode()).toMatchSnapshot();
  });

  it('should have the cards in the landscape orientation when slot is 1', () => {
    component.setProps({slot: 1});

    expect(component.getNode()).toMatchSnapshot();
  });

  it('should have the cards in the portrait orientation when slot is 2', () => {
    component.setProps({slot: 2});

    expect(component.getNode()).toMatchSnapshot();
  });

  it('should have the cards in the landscape orientation when slot is 3', () => {
    component.setProps({slot: 3});

    expect(component.getNode()).toMatchSnapshot();
  });

  it('should display all the taken cards', () => {
    player.takenCards = [[TestCards.create()]];

    component.setProps({player});

    expect(component.getNode()).toMatchSnapshot();
  });

  describe('when a card is clicked', () => {
    let card;

    beforeEach(() => {
      card = component.find('.Player__hand CardComponent').at(1);
    });

    it('should call the onCardClicked prop with the cards index', () => {
      card.simulate('click');

      expect(cardClickedSpy).toHaveBeenCalledWith(1);
    });

    it('should not call the onCardClicked prop if the player is not user controlled', () => {
      player.isUserControlled = false;
      component.setProps({player});

      card.simulate('click');

      expect(cardClickedSpy).not.toHaveBeenCalled();
    });

    it('should not call the onCardClicked prop if the player is not active', () => {
      player.isActive = false;
      component.setProps({player});

      card.simulate('click');

      expect(cardClickedSpy).not.toHaveBeenCalled();
    });
  });
});
