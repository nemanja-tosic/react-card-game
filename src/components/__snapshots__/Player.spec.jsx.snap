// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlayerComponent should be marked as active when active 1`] = `
<div
  className="Player Player--0"
>
  <div
    className="Player__info"
  >
    <div
      className="Player__avatar Player__avatar--0"
    />
    <div
      className="Player__name Player__name--active"
    >
      John
    </div>
    <div
      className="Player__taken-cards"
    >
      
    </div>
  </div>
  <div
    className="Player__hand"
  >
    <div
      className="Player__hand__card"
    >
      <CardComponent
        card={
          Card {
            "code": "0C",
            "imageUrl": "https://deckofcardsapi.com/static/img/0C.png",
          }
        }
        cardRef={[Function]}
        faceDown={false}
        onClick={[Function]}
        orientation="portrait"
      />
    </div>
    <div
      className="Player__hand__card"
    >
      <CardComponent
        card={
          Card {
            "code": "2C",
            "imageUrl": "https://deckofcardsapi.com/static/img/2C.png",
          }
        }
        cardRef={[Function]}
        faceDown={false}
        onClick={[Function]}
        orientation="portrait"
      />
    </div>
    <div
      className="Player__hand__card"
    >
      <CardComponent
        card={
          Card {
            "code": "3C",
            "imageUrl": "https://deckofcardsapi.com/static/img/3C.png",
          }
        }
        cardRef={[Function]}
        faceDown={false}
        onClick={[Function]}
        orientation="portrait"
      />
    </div>
    <div
      className="Player__hand__card"
    >
      <CardComponent
        card={
          Card {
            "code": "4C",
            "imageUrl": "https://deckofcardsapi.com/static/img/4C.png",
          }
        }
        cardRef={[Function]}
        faceDown={false}
        onClick={[Function]}
        orientation="portrait"
      />
    </div>
    <div
      className="Player__hand__card"
    >
      <CardComponent
        card={
          Card {
            "code": "5C",
            "imageUrl": "https://deckofcardsapi.com/static/img/5C.png",
          }
        }
        cardRef={[Function]}
        faceDown={false}
        onClick={[Function]}
        orientation="portrait"
      />
    </div>
  </div>
</div>
`;

exports[`PlayerComponent should be marked as inactive when inactive 1`] = `
<div
  className="Player Player--0"
>
  <div
    className="Player__info"
  >
    <div
      className="Player__avatar Player__avatar--0"
    />
    <div
      className="Player__name Player__name--inactive"
    >
      John
    </div>
    <div
      className="Player__taken-cards"
    >
      
    </div>
  </div>
  <div
    className="Player__hand"
  >
    <div
      className="Player__hand__card"
    >
      <CardComponent
        card={
          Card {
            "code": "0C",
            "imageUrl": "https://deckofcardsapi.com/static/img/0C.png",
          }
        }
        cardRef={[Function]}
        faceDown={false}
        onClick={[Function]}
        orientation="portrait"
      />
    </div>
    <div
      className="Player__hand__card"
    >
      <CardComponent
        card={
          Card {
            "code": "2C",
            "imageUrl": "https://deckofcardsapi.com/static/img/2C.png",
          }
        }
        cardRef={[Function]}
        faceDown={false}
        onClick={[Function]}
        orientation="portrait"
      />
    </div>
    <div
      className="Player__hand__card"
    >
      <CardComponent
        card={
          Card {
            "code": "3C",
            "imageUrl": "https://deckofcardsapi.com/static/img/3C.png",
          }
        }
        cardRef={[Function]}
        faceDown={false}
        onClick={[Function]}
        orientation="portrait"
      />
    </div>
    <div
      className="Player__hand__card"
    >
      <CardComponent
        card={
          Card {
            "code": "4C",
            "imageUrl": "https://deckofcardsapi.com/static/img/4C.png",
          }
        }
        cardRef={[Function]}
        faceDown={false}
        onClick={[Function]}
        orientation="portrait"
      />
    </div>
    <div
      className="Player__hand__card"
    >
      <CardComponent
        card={
          Card {
            "code": "5C",
            "imageUrl": "https://deckofcardsapi.com/static/img/5C.png",
          }
        }
        cardRef={[Function]}
        faceDown={false}
        onClick={[Function]}
        orientation="portrait"
      />
    </div>
  </div>
</div>
`;

exports[`PlayerComponent should display all the taken cards 1`] = `
<div
  className="Player Player--0"
>
  <div
    className="Player__info"
  >
    <div
      className="Player__avatar Player__avatar--0"
    />
    <div
      className="Player__name Player__name--active"
    >
      John
    </div>
    <div
      className="Player__taken-cards"
    >
      <div
        className="Player__taken-cards-group"
      >
        <CardComponent
          card={
            Card {
              "code": "JS",
              "imageUrl": "https://deckofcardsapi.com/static/img/JS.png",
            }
          }
          cardRef={[Function]}
          faceDown={true}
          orientation="portrait"
        />
      </div>
    </div>
  </div>
  <div
    className="Player__hand"
  >
    <div
      className="Player__hand__card"
    >
      <CardComponent
        card={
          Card {
            "code": "0C",
            "imageUrl": "https://deckofcardsapi.com/static/img/0C.png",
          }
        }
        cardRef={[Function]}
        faceDown={false}
        onClick={[Function]}
        orientation="portrait"
      />
    </div>
    <div
      className="Player__hand__card"
    >
      <CardComponent
        card={
          Card {
            "code": "2C",
            "imageUrl": "https://deckofcardsapi.com/static/img/2C.png",
          }
        }
        cardRef={[Function]}
        faceDown={false}
        onClick={[Function]}
        orientation="portrait"
      />
    </div>
    <div
      className="Player__hand__card"
    >
      <CardComponent
        card={
          Card {
            "code": "3C",
            "imageUrl": "https://deckofcardsapi.com/static/img/3C.png",
          }
        }
        cardRef={[Function]}
        faceDown={false}
        onClick={[Function]}
        orientation="portrait"
      />
    </div>
    <div
      className="Player__hand__card"
    >
      <CardComponent
        card={
          Card {
            "code": "4C",
            "imageUrl": "https://deckofcardsapi.com/static/img/4C.png",
          }
        }
        cardRef={[Function]}
        faceDown={false}
        onClick={[Function]}
        orientation="portrait"
      />
    </div>
    <div
      className="Player__hand__card"
    >
      <CardComponent
        card={
          Card {
            "code": "5C",
            "imageUrl": "https://deckofcardsapi.com/static/img/5C.png",
          }
        }
        cardRef={[Function]}
        faceDown={false}
        onClick={[Function]}
        orientation="portrait"
      />
    </div>
  </div>
</div>
`;

exports[`PlayerComponent should have the cards in the landscape orientation when slot is 1 1`] = `
<div
  className="Player Player--1"
>
  <div
    className="Player__info"
  >
    <div
      className="Player__avatar Player__avatar--1"
    />
    <div
      className="Player__name Player__name--active"
    >
      John
    </div>
    <div
      className="Player__taken-cards"
    >
      
    </div>
  </div>
  <div
    className="Player__hand"
  >
    <div
      className="Player__hand__card"
    >
      <CardComponent
        card={
          Card {
            "code": "0C",
            "imageUrl": "https://deckofcardsapi.com/static/img/0C.png",
          }
        }
        cardRef={[Function]}
        faceDown={false}
        onClick={[Function]}
        orientation="landscape"
      />
    </div>
    <div
      className="Player__hand__card"
    >
      <CardComponent
        card={
          Card {
            "code": "2C",
            "imageUrl": "https://deckofcardsapi.com/static/img/2C.png",
          }
        }
        cardRef={[Function]}
        faceDown={false}
        onClick={[Function]}
        orientation="landscape"
      />
    </div>
    <div
      className="Player__hand__card"
    >
      <CardComponent
        card={
          Card {
            "code": "3C",
            "imageUrl": "https://deckofcardsapi.com/static/img/3C.png",
          }
        }
        cardRef={[Function]}
        faceDown={false}
        onClick={[Function]}
        orientation="landscape"
      />
    </div>
    <div
      className="Player__hand__card"
    >
      <CardComponent
        card={
          Card {
            "code": "4C",
            "imageUrl": "https://deckofcardsapi.com/static/img/4C.png",
          }
        }
        cardRef={[Function]}
        faceDown={false}
        onClick={[Function]}
        orientation="landscape"
      />
    </div>
    <div
      className="Player__hand__card"
    >
      <CardComponent
        card={
          Card {
            "code": "5C",
            "imageUrl": "https://deckofcardsapi.com/static/img/5C.png",
          }
        }
        cardRef={[Function]}
        faceDown={false}
        onClick={[Function]}
        orientation="landscape"
      />
    </div>
  </div>
</div>
`;

exports[`PlayerComponent should have the cards in the landscape orientation when slot is 3 1`] = `
<div
  className="Player Player--3"
>
  <div
    className="Player__info"
  >
    <div
      className="Player__avatar Player__avatar--3"
    />
    <div
      className="Player__name Player__name--active"
    >
      John
    </div>
    <div
      className="Player__taken-cards"
    >
      
    </div>
  </div>
  <div
    className="Player__hand"
  >
    <div
      className="Player__hand__card"
    >
      <CardComponent
        card={
          Card {
            "code": "0C",
            "imageUrl": "https://deckofcardsapi.com/static/img/0C.png",
          }
        }
        cardRef={[Function]}
        faceDown={false}
        onClick={[Function]}
        orientation="landscape"
      />
    </div>
    <div
      className="Player__hand__card"
    >
      <CardComponent
        card={
          Card {
            "code": "2C",
            "imageUrl": "https://deckofcardsapi.com/static/img/2C.png",
          }
        }
        cardRef={[Function]}
        faceDown={false}
        onClick={[Function]}
        orientation="landscape"
      />
    </div>
    <div
      className="Player__hand__card"
    >
      <CardComponent
        card={
          Card {
            "code": "3C",
            "imageUrl": "https://deckofcardsapi.com/static/img/3C.png",
          }
        }
        cardRef={[Function]}
        faceDown={false}
        onClick={[Function]}
        orientation="landscape"
      />
    </div>
    <div
      className="Player__hand__card"
    >
      <CardComponent
        card={
          Card {
            "code": "4C",
            "imageUrl": "https://deckofcardsapi.com/static/img/4C.png",
          }
        }
        cardRef={[Function]}
        faceDown={false}
        onClick={[Function]}
        orientation="landscape"
      />
    </div>
    <div
      className="Player__hand__card"
    >
      <CardComponent
        card={
          Card {
            "code": "5C",
            "imageUrl": "https://deckofcardsapi.com/static/img/5C.png",
          }
        }
        cardRef={[Function]}
        faceDown={false}
        onClick={[Function]}
        orientation="landscape"
      />
    </div>
  </div>
</div>
`;

exports[`PlayerComponent should have the cards in the portrait orientation when slot is 0 1`] = `
<div
  className="Player Player--0"
>
  <div
    className="Player__info"
  >
    <div
      className="Player__avatar Player__avatar--0"
    />
    <div
      className="Player__name Player__name--active"
    >
      John
    </div>
    <div
      className="Player__taken-cards"
    >
      
    </div>
  </div>
  <div
    className="Player__hand"
  >
    <div
      className="Player__hand__card"
    >
      <CardComponent
        card={
          Card {
            "code": "0C",
            "imageUrl": "https://deckofcardsapi.com/static/img/0C.png",
          }
        }
        cardRef={[Function]}
        faceDown={false}
        onClick={[Function]}
        orientation="portrait"
      />
    </div>
    <div
      className="Player__hand__card"
    >
      <CardComponent
        card={
          Card {
            "code": "2C",
            "imageUrl": "https://deckofcardsapi.com/static/img/2C.png",
          }
        }
        cardRef={[Function]}
        faceDown={false}
        onClick={[Function]}
        orientation="portrait"
      />
    </div>
    <div
      className="Player__hand__card"
    >
      <CardComponent
        card={
          Card {
            "code": "3C",
            "imageUrl": "https://deckofcardsapi.com/static/img/3C.png",
          }
        }
        cardRef={[Function]}
        faceDown={false}
        onClick={[Function]}
        orientation="portrait"
      />
    </div>
    <div
      className="Player__hand__card"
    >
      <CardComponent
        card={
          Card {
            "code": "4C",
            "imageUrl": "https://deckofcardsapi.com/static/img/4C.png",
          }
        }
        cardRef={[Function]}
        faceDown={false}
        onClick={[Function]}
        orientation="portrait"
      />
    </div>
    <div
      className="Player__hand__card"
    >
      <CardComponent
        card={
          Card {
            "code": "5C",
            "imageUrl": "https://deckofcardsapi.com/static/img/5C.png",
          }
        }
        cardRef={[Function]}
        faceDown={false}
        onClick={[Function]}
        orientation="portrait"
      />
    </div>
  </div>
</div>
`;

exports[`PlayerComponent should have the cards in the portrait orientation when slot is 2 1`] = `
<div
  className="Player Player--2"
>
  <div
    className="Player__info"
  >
    <div
      className="Player__avatar Player__avatar--2"
    />
    <div
      className="Player__name Player__name--active"
    >
      John
    </div>
    <div
      className="Player__taken-cards"
    >
      
    </div>
  </div>
  <div
    className="Player__hand"
  >
    <div
      className="Player__hand__card"
    >
      <CardComponent
        card={
          Card {
            "code": "0C",
            "imageUrl": "https://deckofcardsapi.com/static/img/0C.png",
          }
        }
        cardRef={[Function]}
        faceDown={false}
        onClick={[Function]}
        orientation="portrait"
      />
    </div>
    <div
      className="Player__hand__card"
    >
      <CardComponent
        card={
          Card {
            "code": "2C",
            "imageUrl": "https://deckofcardsapi.com/static/img/2C.png",
          }
        }
        cardRef={[Function]}
        faceDown={false}
        onClick={[Function]}
        orientation="portrait"
      />
    </div>
    <div
      className="Player__hand__card"
    >
      <CardComponent
        card={
          Card {
            "code": "3C",
            "imageUrl": "https://deckofcardsapi.com/static/img/3C.png",
          }
        }
        cardRef={[Function]}
        faceDown={false}
        onClick={[Function]}
        orientation="portrait"
      />
    </div>
    <div
      className="Player__hand__card"
    >
      <CardComponent
        card={
          Card {
            "code": "4C",
            "imageUrl": "https://deckofcardsapi.com/static/img/4C.png",
          }
        }
        cardRef={[Function]}
        faceDown={false}
        onClick={[Function]}
        orientation="portrait"
      />
    </div>
    <div
      className="Player__hand__card"
    >
      <CardComponent
        card={
          Card {
            "code": "5C",
            "imageUrl": "https://deckofcardsapi.com/static/img/5C.png",
          }
        }
        cardRef={[Function]}
        faceDown={false}
        onClick={[Function]}
        orientation="portrait"
      />
    </div>
  </div>
</div>
`;
